malloc/malloc.c

```diff
@@ -23,10 +30,19 @@ char *string_dup(char *src)
     pointers to char pointers first before performing the copying. 
     
     Do not use the `memcpy` function from the standard library.
+
+    https://stackoverflow.com/questions/36931022/what-does-char-0-mean
 */
 void *mem_copy(void *dest, const void *src, int n)
 {
+    // make pointers
+    char *pdest = (char *)dest;
+    char *psrc = (char *)src;  // what is (char *)?
```

Comment:
To my understanding, we're just using chars so we can copy data byte by byte, since the size of a char is 1 byte. The data being copied may not necessarily be characters, but we can cast them that way to ease copying.